pkg/controller/volume/attachdetach/cache/actual_state_of_world.go

```diff
@@ -656,16 +671,13 @@ func (asw *actualStateOfWorld) GetVolumesToReportAttached() map[types.NodeName][
 	defer asw.Unlock()
 
 	volumesToReportAttached := make(map[types.NodeName][]v1.AttachedVolume)
-	for nodeName, nodeToUpdateObj := range asw.nodesToUpdateStatusFor {
+	for _, nodeToUpdateObj := range asw.nodesToUpdateStatusFor {
 		if nodeToUpdateObj.statusUpdateNeeded {
 			volumesToReportAttached[nodeToUpdateObj.nodeName] = asw.getAttachedVolumeFromUpdateObject(nodeToUpdateObj.volumesToReportAsAttached)
 		}
 		// When GetVolumesToReportAttached is called by node status updater, the current status
```

Comment:
Update this comment to say that SetNodeStatusUpdateNotNeeded is expected to be called when the status is updated.